src/main/java/com/example/gymApp/controller/TraineeController.java

```diff
@@ -0,0 +1,357 @@
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/protected/trainee")
+public class TraineeController {
+
+  private static final Logger log = LoggerFactory.getLogger(TraineeController.class);
+  private final TraineeService traineeService;
+  private final UserService userService;
+  private final ProfileService profileService;
+  private final TraineeDto traineeDto;
+  private final TraineeMapper traineeMapper;
+  private final TrainerMapper trainerMapper;
+  private final TrainerService trainerService;
+  private final TrainingService trainingService;
+
+  public TraineeController(TraineeService traineeService, UserService userService,
+      ProfileService profileService, TraineeDto traineeDto,
+      TraineeMapper traineeMapper, TrainerMapper trainerMapper, TrainerService trainerService,
+      TrainingService trainingService) {
+    this.traineeService = traineeService;
+    this.userService = userService;
+    this.profileService = profileService;
+    this.traineeDto = traineeDto;
+    this.traineeMapper = traineeMapper;
+    this.trainerMapper = trainerMapper;
+
+    this.trainerService = trainerService;
+    this.trainingService = trainingService;
+  }
+
+  @GetMapping("/getAll")
+  public List<Trainee> getAllTrainees() {
+    return traineeService.getAllTrainees();
+  }
+
+
+
+  /*
+
+  5. Get Trainee Profile (GET method)
+a. Request
+I. Username (required)
+b. Response
+I. First Name
+II. Last Name
+III. Date of Birth
+IV. Address
+V. Is Active
+VI. Trainers List
+1. Trainer Username
+2. Trainer First Name
+3. Trainer Last Name
+4. Trainer Specialization (Training type reference)
+   */
+
+  @GetMapping("/getTraineeWithTrainersList")
+  public ResponseEntity<?> getTraineeProfileWithTrainersList(@RequestParam String username) {
+
+    Trainee trainee = traineeService.getTraineeByUsername(username);
+
+    List<TrainerDto> trainersList = trainerService.getAlltrainersByTrainee(username).stream()
+        .map(trainer -> new TrainerDto(
+            trainer.getUser().getUsername(),
+            trainer.getUser().getFirstName(),
+            trainer.getUser().getLastName(),
+            trainer.getSpecialization().getName()))
```

Comment:
you can use `org.springframework.core.convert.converter.Converter`

Reply:
done